### ISSUE TITLE
"Jump to" channel switcher

### ISSUE THREAD

**Issue by ssk2** [thelounge/thelounge#1454]
It'd be nice to have a pop box that allows users to quickly switch to a DM or channel with a couple of characters, a la the "Jump to" modal in Slack. 

(In Slack, cmd + k on a Mac will pop up the modal, you can start typing the name of the channel and it will use fuzzy logic to match to a channel.)

**Comment by dgw** [comment]
I was going to suggest a `/goto` command, like what Textual does, but this is far better UX. :+1:

**Comment by richrd** [comment]
This would be really nice! I also suggest that it would sort by highlights and unread messages. Preferably the channel/pm with the latest unread message at the top.
 
For reference this is how it looks in slack: https://get.slack.help/hc/article_attachments/360000957623/Screen_Shot_2018-03-21_at_11.24.49_AM.png

**Comment by teh-mouses** [comment]
This would be amazing and something we really need to add.

**Comment by richrd** [comment]
I started prototyping this now that vue is in master. Any suggestions for which key binding to use? I was thinking <kbd>Alt</kbd>+<kbd>J</kbd> for "jump".

**Comment by Snuupy** [comment]
Sounds like we also need a shortcut key configuration page ;)

**Comment by richrd** [comment]
I've implemented this in https://github.com/thelounge/thelounge/tree/richrd/jump-to
It still need polishing but it's now ready for testing.

**Comment by Snuupy** [comment]
On macOS, pressing Alt-J inserts a symbol in the jump-to switcher: 
![image](https://user-images.githubusercontent.com/964538/70861546-6f3c3f80-1efd-11ea-8c82-c4df84a47e77.png)

**Comment by richrd** [comment]
@Snuupy does that happen the first time you open the Jump to box, or only after pressing opt+j twice?
Apparently opt+j is a defaut mac shortcut for ∆.

**Comment by Snuupy** [comment]
@richrd First time. As a comparison, opt+a does not insert 'å' and switches channels accordingly.

**Comment by richrd** [comment]
This feature is now moved into the sidebar and ready for testing and comments in PR  #3594
